### ISSUE TITLE
[bug]  -R Failed to set filetime, -o file name has  Full-width character

### ISSUE THREAD

**Issue by tec1987** [curl/curl#5658]
### I did this
```
D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -O
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    172      0  0:00:04  0:00:04 --:--:--   172

D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -o cc.asc
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    398      0  0:00:02  0:00:02 --:--:--   398

D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -o 校验.asc
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    509      0  0:00:01  0:00:01 --:--:--   509
Failed to set filetime 1593598063 on outfile: CreateFile failed: GetLastError 2

D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -o ".\temp\校验.asc"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    130      0  0:00:06  0:00:06 --:--:--   201
Failed to set filetime 1593598063 on outfile: CreateFile failed: GetLastError 2

D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -o "ａｂｃ.asc"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    698      0  0:00:01  0:00:01 --:--:--   698
Failed to set filetime 1593598063 on outfile: CreateFile failed: GetLastError 123

D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -o "＄Ｔ＃.asc"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    802      0  0:00:01  0:00:01 --:--:--   802
Failed to set filetime 1593598063 on outfile: CreateFile failed: GetLastError 123

D:\>curl https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -o "ＴＥＳＴ.asc"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0    812      0  0:00:01  0:00:01 --:--:--   812
Failed to set filetime 1593598063 on outfile: CreateFile failed: GetLastError 2
```
All above file is download, only Failed to set filetime.

Use old version on XP without this problem. 

### I expected the following

Use this：
https://sslsites.de/paehl.com/open_source/?CURL_7.71.0
and this：
https://skanthak.homepage.t-online.de/curl.html

build no this problem. 

### curl/libcurl version

[curl -V output]
> curl 7.71.1 (x86_64-pc-win32) libcurl/7.71.1 OpenSSL/1.1.1g (Schannel) zlib/1.2.11 brotli/1.0.7 WinIDN libssh2/1.9.0 nghttp2/1.41.0
> Release-Date: 2020-07-01
> Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
> Features: AsynchDNS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile MultiSSL NTLM SPNEGO SSL SSPI TLS-SRP UnixSockets brotli libz

### operating system

Windows 10 Enterprise 2016 LTSB 1607 x64

**Comment by jay** [comment]
The setfiletime function uses CreateFileA. There were Unicode changes made to the curl tool recently and I suspect that may have something to do with it. Probably the filename is UTF-8 encoded now? @MarcelRaad

https://github.com/curl/curl/blob/5a1fc8d33808d7b22f57bdf9403cda7ff07b0670/src/tool_filetime.c#L106-L129

**Comment by MarcelRaad** [comment]
Thanks for the report, and sorry for noticing it so late! Should be fixed with #5718 . Now I get:
```
PS C:\curl\build\Win64\VC15\LIB Debug - DLL Windows SSPI - DLL WinIDN> .\curld.exe https://dl.bintray.com/vszakats/generic/curl-7.71.1-win32-mingw.tar.xz.asc -R -O
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   821  100   821    0     0   1319      0 --:--:-- --:--:-- --:--:--  1328
```

**Comment by tec1987** [comment]
Now is work fine.
Thanks for all.
